gcalbot/gcalbot/accounts.go

```diff
@@ -0,0 +1,148 @@
+package gcalbot
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kballard/go-shellquote"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/chat1"
+	"github.com/keybase/managed-bots/base"
+)
+
+func (h *Handler) HandleAuth(msg chat1.MsgSummary) error {
+	cmd := strings.TrimSpace(msg.Content.Text.Body)
+	toks, err := shellquote.Split(cmd)
+	if err != nil {
+		h.ChatDebug(msg.ConvID, "error splitting command: %s", err)
+		return nil
+	}
+	args := toks[3:]
+	if len(args) != 1 {
+		h.ChatDebug(msg.ConvID, "bad args for accounts connect: %s", args)
+		return nil
+	}
+
+	username := msg.Sender.Username
+	accountNickname := args[0]
+	err = h.db.InsertAccountForUser(username, accountNickname)
+	if err != nil {
+		h.ChatDebug(msg.ConvID, "error connecting account %s", accountNickname)
+		return err
+	}
+
+	_, err = h.kbc.SendMessageByConvID(msg.ConvID, "Account '%s' has been connected.", accountNickname)
+	return err
+}
+
+func (h *Handler) accountsListHandler(msg chat1.MsgSummary) error {
+	username := msg.Sender.Username
+	accounts, err := h.db.GetAccountsForUser(username)
+	if err != nil {
+		h.ChatDebug(msg.ConvID, "error fetching accounts from database %q", err)
+		return nil
+	}
+
+	if len(accounts) == 0 {
+		_, err = h.kbc.SendMessageByConvID(msg.ConvID, "You have no connected accounts.")
+		if err != nil {
+			return fmt.Errorf("error sending message: %s", err)
+		}
+		return nil
+	}
+
+	accountListMessage := "Here are your connected accounts:" + strings.Repeat("\n• %s", len(accounts))
+	_, err = h.kbc.SendMessageByConvID(msg.ConvID, accountListMessage, accounts...)
+	if err != nil {
+		return fmt.Errorf("error sending message: %s", err)
+	}
+	return nil
+}
+
+func (h *Handler) accountsConnectHandler(msg chat1.MsgSummary, args []string) error {
+	if len(args) != 1 {
+		_, err := h.kbc.SendMessageByConvID(msg.ConvID, "Invalid number of arguments.")
+		if err != nil {
+			return fmt.Errorf("error sending message: %s", err)
+		}
+		return nil
+	}
+
+	username := msg.Sender.Username
+	accountNickname := args[0]
+
+	exists, err := h.db.ExistsAccountForUser(username, accountNickname)
+	if err != nil {
+		return fmt.Errorf("error checking for account: %s", err)
+	}
+	if exists {
```

Comment:
if this is true you can probably just short circuit with out the messaging

Reply:
As in just return? If I return `nil` the user won't get any feedback, if I return an error the user will get `Something went wrong!`

Reply:
Nothing is wrong in this case though, if the account already exists there is nothing to do so it's not helpful to return an error

Reply:
It's not returning an error, it's just letting the user know that an account with that nickname is already connected:
```
An account connection with the nickname '%s' already exists.
```
If you don't think that it's necessary to provide any feedback, I can drop this block

Reply:
No feedback necessary here I think

Reply:
So should the account be overwritten or should it just not do anything?